src/back-to-top/index.jsx

```diff
@@ -0,0 +1,49 @@
+import React, { Component } from 'react';
+import classnames from 'classnames';
+
+class BackToTop extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isVisible: this.props.isVisible || false
+    };
+
+    this.handleScroll = this.handleScroll.bind(this);
+  }
+
+  componentDidMount() {
+    window.addEventListener('scroll', this.handleScroll, true);
+    this.handleScroll();
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('scroll', this.handleOnScroll);
+  }
+
+  handleScroll() {
+    if (!this.state.isVisible && window.pageYOffset > this.props.showAt) {
+      this.setState({ isVisible: true });
+    } else if (this.state.isVisible && window.pageYOffset < this.props.showAt) {
+      this.setState({ isVisible: false });
+    }
+  }
+
+  scrollToTop() {
+    window.scrollTo(0, 0);
```

Comment:
this is an instant jump - you can pass it behaviour: smooth for a nice animated scroll
```js
window.scrollTo({
  top: 0,
  left: 0,
  behavior: 'smooth'
});
```